web-wallet/src/lib/url/__tests__/makeNodeUrl.spec.js

```diff
@@ -0,0 +1,155 @@
+import { makeNodeUrl } from "..";
+// import { JSDOM } from "jsdom";
+
+describe("makeNodeUrl", () => {
+  const localhostString = window.location.hostname;
+
+  it("should return a local URL when VITE_NODE_URL is not set", () => {
+    delete import.meta.env.VITE_NODE_URL;
+    expect(makeNodeUrl().hostname).toBe(localhostString);
+  });
+
+  it("should return a local URL when VITE_NODE_URL is an empty string", () => {
+    import.meta.env.VITE_NODE_URL = "";
+    expect(makeNodeUrl().hostname).toBe(localhostString);
+  });
+
+  it("should return a local URL with no base path when VITE_NODE_URL is not set and VITE_RUSK_PATH is not set", () => {
+    delete import.meta.env.VITE_NODE_URL;
+    delete import.meta.env.VITE_RUSK_PATH;
+    expect(makeNodeUrl().hostname).toBe(localhostString);
+    expect(makeNodeUrl().pathname).toBe("/");
+  });
+
+  it("should return a local URL with no base path when VITE_NODE_URL is set to an empty string and VITE_RUSK_PATH is not set", () => {
+    import.meta.env.VITE_NODE_URL = "";
+    delete import.meta.env.VITE_RUSK_PATH;
+    expect(makeNodeUrl().hostname).toBe(localhostString);
+    expect(makeNodeUrl().pathname).toBe("/");
+  });
+
+  it("should return a local URL with no base path when VITE_NODE_URL is not set and VITE_RUSK_PATH is set to an empty string", () => {
+    delete import.meta.env.VITE_NODE_URL;
+    import.meta.env.VITE_RUSK_PATH = "";
+    expect(makeNodeUrl().hostname).toBe(localhostString);
+    expect(makeNodeUrl().pathname).toBe("/");
+  });
+
+  it("should return a local URL with no base path when VITE_NODE_URL is set to an empty string and VITE_RUSK_PATH is set to an empty string", () => {
+    import.meta.env.VITE_NODE_URL = "";
+    import.meta.env.VITE_RUSK_PATH = "";
+    expect(makeNodeUrl().hostname).toBe(localhostString);
+    expect(makeNodeUrl().pathname).toBe("/");
+  });
+
+  it("should return a local URL with a base path when `VITE_NODE_URL` is not set and `VITE_RUSK_PATH` is set to a valid string", () => {
+    delete import.meta.env.VITE_NODE_URL;
+    import.meta.env.VITE_RUSK_PATH = "/testing";
+    expect(makeNodeUrl().hostname).toBe(localhostString);
+    expect(makeNodeUrl().pathname).toBe("/testing");
+  });
+
+  it("should return a local URL with a base path when `VITE_NODE_URL` is set to an empty string and `VITE_RUSK_PATH` is set to a valid string", () => {
+    import.meta.env.VITE_NODE_URL = "";
+    import.meta.env.VITE_RUSK_PATH = "/testing";
+    expect(makeNodeUrl().hostname).toBe(localhostString);
+    expect(makeNodeUrl().pathname).toBe("/testing");
+  });
+
+  it("should return the devnet URL when the hostname starts with 'devnet.staging' on the staging URL", () => {
+    global.window = Object.create(window);
+
+    const url = new URL("https://devnet.staging.dusk.network");
+
+    Object.defineProperty(window, "location", {
+      value: {
+        hostname: url.hostname,
+        href: url.href,
+        protocol: url.protocol,
+      },
+    });
+
```

Comment:
This is very odd.
Anyway instead of repeating this code for each test, just create a spy on `window.location` and make its return value what you want for each test.

Reply:
It's odd, but it's a workaround from the Jest community to create.  I will optimised it, though.